File: modules/app-version/ios/AppVersionModule.swift
import ExpoModulesCore

public class AppVersionModule: Module {
  public func definition() -> ModuleDefinition {
    Name("AppVersion")

    Function("getVersion") {
      return Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? "unknown"
    }
  }
}

File: modules/app-version/android/src/main/java/expo/modules/appversion/AppVersionModule.kt
package expo.modules.appversion

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class AppVersionModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("AppVersion")

    Function("getVersion") {
      appContext.reactContext?.packageManager
        ?.getPackageInfo(appContext.reactContext!!.packageName, 0)
        ?.versionName ?: "unknown"
    }
  }
}
